### PR TITLE
hide prim2orig in executor

### PR DESCRIPTION
**Pull request by levi131**
### PR types
Others

### PR changes
Others

### Describe
The conversion function from automatic differential basic operator to native operator is integrated into the executor

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/autograd/test_primapi.py [hunk 1]
```diff
@@ -23,6 +23,117 @@
...
+        paddle.enable_static()
+        paddle.incubate.autograd.enable_prim()
+
+    def tearDown(self):
+        paddle.incubate.autograd.disable_prim()
+        paddle.disable_static()
+
+    def test_forward_grad_without_program_guard(self):
+
+        def with_param_guard():
```
Comment (cxxly) [inline python/paddle/fluid/tests/unittests/autograd/test_primapi.py]: with_program_guard ?

File: python/paddle/fluid/tests/unittests/autograd/test_primapi.py [hunk 2]
```diff
@@ -23,6 +23,117 @@
...
+                ys_grad = paddle.incubate.autograd.forward_grad(
+                    ys, static_xs, static_v)
+                paddle.incubate.autograd.prim2orig(mp.block(0))
+            exe = paddle.static.Executor()
+            exe.run(sp)
+            out = exe.run(mp, feed=feed, fetch_list=ys_grad)
+            paddle.incubate.autograd.disable_prim()
+            return out
+
+        def without_param_guard():
```
Comment (cxxly) [inline python/paddle/fluid/tests/unittests/autograd/test_primapi.py]: without_program_guard

File: python/paddle/fluid/tests/unittests/autograd/test_primapi.py [hunk 3]
```diff
@@ -23,6 +23,117 @@
...
+        actual = without_param_guard()
+        self.assertEqual(type(actual), type(expected))
+        np.testing.assert_allclose(np.concatenate(actual),
+                                   np.concatenate(expected),
+                                   rtol=self._rtol,
+                                   atol=self._atol)
+
+    def test_grad_without_program_guard(self):
+
+        def with_param_guard():
```
Comment (cxxly) [inline python/paddle/fluid/tests/unittests/autograd/test_primapi.py]: 同上

File: python/paddle/fluid/tests/unittests/autograd/test_primapi.py [hunk 4]
```diff
@@ -23,6 +23,117 @@
...
+                    static_xs, typing.Sequence) else self.fun(static_xs)
+                xs_grad = paddle.incubate.autograd.grad(ys, static_xs, static_v)
+                paddle.incubate.autograd.prim2orig(mp.block(0))
+            exe = paddle.static.Executor()
+            exe.run(sp)
+            out = exe.run(mp, feed=feed, fetch_list=xs_grad)
+            paddle.incubate.autograd.disable_prim()
+            return out
+
+        def without_param_guard():
```
Comment (cxxly) [inline python/paddle/fluid/tests/unittests/autograd/test_primapi.py]: 同上

File: python/paddle/fluid/tests/unittests/autograd/test_primapi.py (hunk 1 above)
Comment (levi131) [reply python/paddle/fluid/tests/unittests/autograd/test_primapi.py]: updated

File: python/paddle/fluid/tests/unittests/autograd/test_primapi.py (hunk 2 above)
Comment (levi131) [reply python/paddle/fluid/tests/unittests/autograd/test_primapi.py]: updated

File: python/paddle/fluid/tests/unittests/autograd/test_primapi.py (hunk 4 above)
Comment (levi131) [reply python/paddle/fluid/tests/unittests/autograd/test_primapi.py]: updated

File: python/paddle/fluid/tests/unittests/autograd/test_primapi.py (hunk 3 above)
Comment (levi131) [reply python/paddle/fluid/tests/unittests/autograd/test_primapi.py]: updated

File: python/paddle/fluid/executor.py [hunk 5]
```diff
@@ -1453,6 +1453,11 @@ def _can_use_interpreter_core(program, place):
                         ir_graph = framework.IrGraph(compiled_graph,
                                                      for_test=True)
                         inner_program = ir_graph.to_program()
+                    else:
+                        from ..incubate.autograd import prim_enabled, prim2orig
```
Comment (From00) [inline python/paddle/fluid/executor.py]: It is suggested to modify this import statement that do not comply with the Python style guide in the following aspects: 
1. Imports should always put at the top of the file, just after any module comments and docstrings and before module globals and constants. 
2. The relative names should not be used in imports. The usage of full package name helps prevent unintentionally importing a package twice.
3. The import statements should only used for packages and modules, not for individual classes or functions.

Comment (levi131) [reply python/paddle/fluid/executor.py]: 1、根据paddle目前的文件路径设置，放在文件起始处有循环依赖问题，选择在函数体内使用处import。
2、可以改为绝对路径

Comment (levi131) [reply python/paddle/fluid/executor.py]: 讨论对齐后使用绝对路径

Comment (cxxly) [reply python/paddle/fluid/executor.py]: > It is suggested to modify this import statement that do not comply with the Python style guide in the following aspects:
> 
> 1. Imports should always put at the top of the file, just after any module comments and docstrings and before module globals and constants.
> 2. The relative names should not be used in imports. The usage of full package name helps prevent unintentionally importing a package twice.
> 3. The import statements should only used for packages and modules, not for individual classes or functions.

非常赞同严格遵循代码规范，Paddle目前存在各种各样导入方式，代码检查工具目前检查不出来。

**Review by From00 — APPROVED** [review]

**Pull request merged**